packages/discovery-provider/src/queries/get_comments.py

```diff
@@ -215,40 +234,45 @@ def get_track_comments(args, track_id, current_user_id=None):
             )
             .offset(offset)
             .limit(limit)
-            .all()
         )
 
-        return [
-            {
-                "id": encode_int_id(track_comment.comment_id),
-                "user_id": (
-                    encode_int_id(track_comment.user_id)
-                    if not track_comment.is_delete
-                    else None
-                ),
-                "message": (
-                    track_comment.text if not track_comment.is_delete else "[Removed]"
-                ),
-                "is_edited": track_comment.is_edited,
-                "track_timestamp_s": track_comment.track_timestamp_s,
-                "react_count": react_count,
-                "reply_count": reply_count,
-                "is_current_user_reacted": get_is_reacted(
-                    session, current_user_id, track_comment.comment_id
-                ),
-                "is_artist_reacted": get_is_reacted(
-                    session, artist_id, track_comment.comment_id
-                ),
-                "replies": get_replies(
-                    session, track_comment.comment_id, current_user_id, artist_id
-                ),
-                "created_at": str(track_comment.created_at),
-                "updated_at": str(track_comment.updated_at),
-                "is_tombstone": track_comment.is_delete and reply_count > 0,
-                "is_muted": is_muted if is_muted is not None else False,
-            }
-            for [track_comment, react_count, reply_count, is_muted] in track_comments
-        ]
+        track_comment_res = []
+        for [track_comment, react_count, is_muted] in track_comments.all():
+            replies = get_replies(
+                session, track_comment.comment_id, current_user_id, artist_id
+            )
+            reply_count = len(replies)
```

Comment:
This reply count won't be correct since it should only be getting the first 3 - I did this on my PR; in order to get the true count I passed `limit=None` to `get_replies()`

That being said, I do wonder if at this point it makes sense to paginate the replies via network call anymore - we need to query for all replies in order to have an accurate count, so maybe we should just nix that API call entirely? We can just show hide the replies on the UI 🤔